test/parallel/test-net-server-close.js

```diff
@@ -26,7 +26,23 @@ const net = require('net');
 
 const sockets = [];
 
-const server = net.createServer(function(c) {
+let server = new net.Server();
+
+// Ensure that the close callback is not called when the server was not
+// previously running.
+server.close(common.mustNotCall());
```

Comment:
This is not correct. We should always call the callback.

Reply:
@mcollina could you please clarify why we need to close the server if that server is not running?

Reply:
FWIW, I agree with Matteo. The callback here should be called in case there is any cleanup logic that is deferred to close (e.g. closing fd's). There's a non-trivial risk of breakage here and I don't see this change as being worth it now that I reconsider.